### PR TITLE
codecov yaml

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

This yaml file configures how codecov comments on pull requests.  It should now "succeed" even when pull requests reduce coverage by up to 0.5%

@rhaschke here is the documentation for this: https://docs.codecov.io/docs/commit-status

### CODE REVIEW

File: codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+        # allow coverage to drop by this amount and still post success
+        threshold: 2%
```
Comment (rhaschke) [inline codecov.yaml]: I think it suffices to limit the threshold to 0.5%, given the size of the code base.
Would it be possible to reduce the severity of a codecov failure (from red cross to yellow dot in github)?

Comment (tylerjw) [reply codecov.yaml]: I wasn't able to find an option for the yellow dot in the configuration.  From my reading of the documentation I am not sure if this is 2% of the content of the existing pr, 2% of the overall code coverage, or both.  If is of the overall coverage than I agree with you that I could be much smaller.

Comment (rhaschke) [reply codecov.yaml]: I guess it's the percentage reported by the codecov reports itself. What I have seen so far, are values like -0.39%. Maybe check some recent PRs to get a better impression of the actual range.

Comment (tylerjw) [reply codecov.yaml]: changed it to 0.5

**Comment by rhaschke** [comment]
Fixes #1896.

**Pull request merged**